tests/thread_float/tests/01-run.py

```diff
@@ -26,25 +26,36 @@ def same_computation_as_in_c_prog(thread_num):
     return f
 
 
+def parse_result_line(child):
+    child.expect(r"t(\d): (\d{3}\.\d+|[+-]*nan|[+-]inf)\r\n")
+    # Note: intentionally keeping the float output as string to also test that
+    # printf("%f", ...) prints the exact same char sequence for the same float
+    # value each time
+    return (int(child.match.group(1)), child.match.group(2))
+
+
 def testfunc(child):
     child.expect_exact("THREADS CREATED")
-    child.expect_exact("THREAD t1 start")
-    child.expect_exact("THREAD t2 start")
-    child.expect_exact("THREAD t3 start")
-
+    threads_started = set()
+    for _ in range(3):
```

Comment:
This could use a comment like

```
        # Threads are started with THREAD_CREATE_WOUT_YIELD, and thus prone starting
        # in a random sequence depending on any interrupts that cause rescheduling.
```